File: boj/1157.cpp
#include <iostream>
#include <string>

using namespace std;

// https://www.acmicpc.net/problem/1157
// gcc version 6.3.0 (MinGW.org GCC-6.3.0-1)

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);

    int arr[26] = { 0 };
    string s;
    int max = 0;
    char answer;

    cin >> s;

    // 문자를 하나씩 비교하여 값 넣기
    for(int i=0; i<s.length(); i++) {
        if(65 <= s[i] && s[i] <= 90)
            arr[s[i] - 65]++;
        else if(97 <= s[i] && s[i] <= 122)
            arr[s[i] - 97]++;
    }

    // 가장 숫자가 큰 문자 찾기
    for(int i=0; i<26; i++) {
        if(arr[i] > max) {
            answer = i + 65;
            max = arr[i];
        } else if(arr[i] == max) {
            answer = '?';
        }
    }

    cout << answer;
}

File: boj/1157.java
import java.io.*;

// https://www.acmicpc.net/problem/1157

class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        int[] arr = new int[26];
        
        int len = s.length();

        // 각 글자 갯수를 arr 배열에 저장
        for(int i=0; i<len; i++) {
            int idx = Character.toLowerCase(s.charAt(i)) - 'a';
            arr[idx]++;
        }

        int max = -1;
        char answer = '?';

        for(int i=0; i<26; i++) {
            if(arr[i] > max) {
                max = arr[i];
                answer = (char) (i+65);
            } else if(arr[i] == max)
                answer = '?';
        }

        System.out.println(answer);
    }
}